locust/contrib/fasthttp.py

```diff
@@ -75,13 +75,18 @@ def __init__(
         user: Optional[User],
         insecure=True,
         client_pool: Optional[HTTPClientPool] = None,
+        ssl_context: Optional[SSLContext] = None,
         **kwargs,
     ):
         self.environment = environment
         self.base_url = base_url
         self.cookiejar = CookieJar()
         self.user = user
-        if insecure:
+        if ssl_context:
+            if not isinstance(ssl_context, SSLContext):
+                raise TypeError("You must provide a valid SSLContext.")
+            ssl_context_factory = lambda: ssl_context
```

Comment:
Doesnt it make more sense to just pass a function instead of an instantiated SSLContext that we make a fake function for? (I made a comment and then deleted it, but after some more thought I think my initial comment was correct :)

Reply:
Surely it does. I was trying to make use of `ssl_context` as an actual context then the caller could set it accordingly, but making it a factory keeps the way it does in the background. I will rename it to `ssl_context_factory` and type-hint it as Callable. Thanks for reviewing it.